examples/fabric-admin/commands/pairing/PairingCommand.cpp

```diff
@@ -564,15 +586,46 @@ void PairingCommand::OnCurrentFabricRemove(void * context, NodeId nodeId, CHIP_E
 
 chip::Optional<uint16_t> PairingCommand::FailSafeExpiryTimeoutSecs() const
 {
-    // We don't need to set additional failsafe timeout as we don't ask the final user if he wants to continue
+    // No manual input, so do not need to extend.
     return chip::Optional<uint16_t>();
 }
 
+bool PairingCommand::ShouldWaitAfterDeviceAttestation()
+{
+    // If there is a vendor ID and product ID, request OnDeviceAttestationCompleted().
+    SetupPayload payload;
+    CHIP_ERROR err = GetPayload(mOnboardingPayload, payload);
+    return err == CHIP_NO_ERROR && payload.vendorID != 0 && payload.productID != 0;
+}
+
 void PairingCommand::OnDeviceAttestationCompleted(chip::Controller::DeviceCommissioner * deviceCommissioner,
                                                   chip::DeviceProxy * device,
                                                   const chip::Credentials::DeviceAttestationVerifier::AttestationDeviceInfo & info,
                                                   chip::Credentials::AttestationVerificationResult attestationResult)
 {
+
+    SetupPayload payload;
+    CHIP_ERROR parse_error = GetPayload(mOnboardingPayload, payload);
+    if (parse_error == CHIP_NO_ERROR && payload.vendorID != 0 && payload.productID != 0)
+    {
+        if (payload.vendorID != info.BasicInformationVendorId() || payload.productID != info.BasicInformationProductId())
+        {
+            ChipLogProgress(NotSpecified,
+                            "Failed validation of vendorID or productID."
+                            "Requested VID: %u, Requested PID: %u,"
+                            "Detected VID: %u, Detected PID %u.",
+                            payload.vendorID, payload.productID, info.BasicInformationVendorId(), info.BasicInformationProductId());
+            deviceCommissioner->ContinueCommissioningAfterDeviceAttestation(
+                device,
+                payload.vendorID == info.BasicInformationVendorId()
+                    ? chip::Credentials::AttestationVerificationResult::kDacProductIdMismatch
+                    : chip::Credentials::AttestationVerificationResult::kDacVendorIdMismatch);
+        }
+
+        deviceCommissioner->CommissioningStageComplete(CHIP_NO_ERROR);
```

Comment:
Are we now supposed to call this because previously we returned true for `ShouldWaitAfterDeviceAttestation`

Reply:
We call this to tell deviceCommissioner that we were successful. 

ContinueCommissioningAfterDeviceAttestation() is poorly named. It will log errors even if it is told that the validation was successful. It seems specifically targeted for the failed, but platform wants to continue case.

Reply:
Could we call this function directly to end the commission process? It looks we need to call deviceCommissioner->ContinueCommissioningAfterDeviceAttestation(
            device, chip::Credentials::AttestationVerificationResult::kSuccess); in success case?

Reply:
ContinueCommissioningAfterDeviceAttestation() appears to be poorly named.  It's success case will log that attestation failed and it is commissioning anyhow. It seems it should have been named ContinueCommissioningAfterDeviceAttestationFailure(). The end result is a call to `CommissioningStageComplete(CHIP_NO_ERROR);`

The end of ContinueCommissioningAfterDeviceAttestation(): 

```
    ChipLogProgress(Controller, "Continuing commissioning after attestation failure for device ID 0x" ChipLogFormatX64,
                    ChipLogValueX64(commissioneeDevice->GetDeviceId()));

    if (attestationResult != AttestationVerificationResult::kSuccess)
    {
        ChipLogError(Controller, "Client selected error: %u for failed 'Attestation Information' for device",
                     to_underlying(attestationResult));

        CommissioningDelegate::CommissioningReport report;
        report.Set<AttestationErrorInfo>(attestationResult);
        CommissioningStageComplete(CHIP_ERROR_INTERNAL, report);
    }
    else
    {
        ChipLogProgress(Controller, "Overriding attestation failure per client and continuing commissioning");
        CommissioningStageComplete(CHIP_NO_ERROR);
    }
```

Reply:
Per API contract by design, when OnDeviceAttestationCompleted is called (attestation failed or 
ShouldWaitAfterDeviceAttestation return true), we always call ContinueCommissioningAfterDeviceAttestation() to proceed, either in success or failure.

In success case, it is equivalent to log some messages and call CommissioningStageComplete(CHIP_NO_ERROR);

Reply:
ack. Let's conform to API spec and call ContinueCommissioningAfterDeviceAttestation(). We'll just have some noisy logs.